### DISCUSSION TITLE
Regex-fueled refactoring

### DISCUSSION THREAD

**Discussion by almirsarajcic** [3Stones-io/omedis#309] [General]
We can post some string replacement cases that helped us do refactoring.
I worry we might lose it otherwise. Usually, I put this stuff in my notes so I can find it later, but it might be useful to others as well.

**Comment by almirsarajcic** [comment]
### Fixtures used without {:ok, _} = pattern matching
I've had issues with some fixtures that were silently failing due to validation and other errors.
To prevent that from happening, we can replace all the cases that have `{:ok, _} =` part missing.

Here's what I'll do:
Replace `(?!\} =)\n\n(\s*)create_` -> `\n\n$1{:ok, _} = create_` in the `test` directory.

<img width="1274" alt="Screenshot 2024-11-09 at 17 37 03" src="https://github.com/user-attachments/assets/4a7553c0-c361-4b2d-b52d-8b22a32bca8e">

**Reply by almirsarajcic** [reply]
Thanks @amos-kibet for the `(?!)` tip 😃

**Comment by almirsarajcic** [comment]
### Resource renaming
We have described the renaming process in detail in https://github.com/wintermeyer/omedis/discussions/284.
There are several Regex patterns we used for that.

**Comment by wintermeyer** [comment]
> We can post some string replacement cases that helped us do refactoring. I worry we might lose it otherwise. Usually, I put this stuff in my notes so I can find it later, but it might be useful to others as well.

Let’s discuss this in our next call. IMO a wiki page is better suited for this than a discussion.
